Download specific versions of plugins

owu-1 opened the pull request:
Download specific versions of plugins so I know what versions the plugins are on. Without knowing that its hard to know if errors happening are because of older versions. Also if I'm reporting errors I need to know the version. Some plugins don't show their build number in their version command output.

- [ ] ~~Don't download versions we already have~~

owu-1 committed:
Use mod versions to download

owu-1 committed:
Simplify by removing old plugins

y2k04 commented on mods.sh:
```diff
@@ -1,6 +1,7 @@
 #!/bin/bash
 mkdir -p plugins
 cd plugins || exit
```
Why is the exit command piped from a change directory command? Was that just for debugging and you forgot to remove it, or what?

owu-1 replied:
Its like if the cd fails then exit the script

owu-1 replied:
I'm linting it with this https://marketplace.visualstudio.com/items?itemName=timonwong.shellcheck
And heres its explaination https://www.shellcheck.net/wiki/SC2164

y2k04 replied:
Cool

y2k04 committed:
Remove "./" in jar removal

We are already in the folder, so it is not needed.

owu-1 reviewed (commented):
Linter doesnt like it without the './' : https://www.shellcheck.net/wiki/SC2035

y2k04 committed:
Remove unnecessary curly brackets

Brackets don't need to be used for variables if they are inside of quotation marks

y2k04 commented:
> Linter doesnt like it without the './' : https://www.shellcheck.net/wiki/SC2035

Ok

y2k04 committed:
Remove modrinth and github api variables

They will never change and are only used inside the function, so why bother putting it in a variable?

y2k04 committed:
Simplify plugin downloads

Moved echo command to each function, as well as downloading the plugin in each function.

y2k04 committed:
Revert changes: ./ in plugin deleting

y2k04 committed:
Fix lint errors

y2k04 committed:
Create variables.sh

y2k04 committed:
Create pluginAPIs.sh

y2k04 committed:
Moved stuff into other scripts

Moved variables into variables.sh
Moved functions that were used to download plugins into pluginAPIs.sh

y2k04 committed:
Marked variables.sh and pluginsAPIs.sh as executable

y2k04 committed:
Add shebang to make lint happy

y2k04 committed:
Added quotation marks to make lint happy

y2k04 committed:
Added quotation marks to make lint happy

y2k04 committed:
Added 'export' command to make lint happy

y2k04 committed:
Added quotation marks to make lint happy

y2k04 committed:
Swapped {} for () in github function to make lint happy

y2k04 committed:
fuck lint

y2k04 committed:
Added shebang

y2k04 committed:
added curly brackets back

fuck lint

y2k04 committed:
fuck lint

y2k04 committed:
Create latestMods.sh

y2k04 committed:
Simplify functions and added latest mods functions

y2k04 committed:
Rename function 'github' to 'ghrelease'

y2k04 committed:
Change functions used

Using latest versions of each function

y2k04 committed:
Added first selector to latest_jenkins

y2k04 committed:
Fix typo

y2k04 committed:
Fixed my fuck ups

y2k04 committed:
Delete latestMods.sh

y2k04 committed:
Added LOADER variable

y2k04 committed:
One last fixup

y2k04 committed:
Grab variables and api functions before changing into plugins directory

y2k04 committed:
Replaced very verbose curl output to be a progress bar!
